client/src/app/shared/components/basic-import-list/basic-import-list.component.ts

```diff
@@ -0,0 +1,365 @@
+
+    /**
+     * Switch that turns true if a file has been selected in the input
+     */
+    public hasFile = false;
+
+    /**
+     * Currently selected encoding. Is set and changed by the config's available
+     * encodings and user mat-select input
+     */
+    public selectedEncoding = 'utf-8';
+
+    /**
+     * indicator on which elements to display
+     */
+    public shown: 'all' | 'error' | 'noerror' = 'all';
+
+    /**
+     * @returns the amount of total item successfully parsed
+     */
+    public get totalCount(): number {
+        return this.importer && this.hasFile ? this.importer.summary.total : null;
+    }
+
+    /**
+     * @returns the encodings available and their labels
+     */
+    public get encodings(): ValueLabelCombination[] {
+        return this.importer.encodings;
+    }
+
+    /**
+     * @returns the available column separators and their labels
+     */
+    public get columnSeparators(): ValueLabelCombination[] {
+        return this.importer.columnSeparators;
+    }
+
+    /**
+     * @eturns the available text separators and their labels
+     */
+    public get textSeparators(): ValueLabelCombination[] {
+        return this.importer.textSeparators;
+    }
+
+    /**
+     * @returns the amount of import items that will be imported
+     */
+    public get newCount(): number {
+        return this.importer && this.hasFile ? this.importer.summary.new : 0;
+    }
+
+    /**
+     * @returns the number of import items that cannot be imported
+     */
+    public get nonImportableCount(): number {
+        if (this.importer && this.hasFile) {
+            return this.importer.summary.errors + this.importer.summary.duplicates;
+        }
+        return 0;
+    }
+
+    /**
+     * @returns the number of import items that have been successfully imported
+     */
+    public get doneCount(): number {
+        return this.importer && this.hasFile ? this.importer.summary.done : 0;
+    }
+
+    private root = this.host.nativeElement;
+
+    public constructor(componentServiceCollector: ComponentServiceCollector, private host: ElementRef<HTMLElement>) {
+        super(componentServiceCollector);
+    }
+
+    /**
+     * Starts with a clean preview (removing any previously existing import previews)
+     */
+    public ngOnInit(): void {
+        this.importer.clearPreview();
+        this.columnSet = this.createColumnSet();
+        this.resetRowHeight();
+    }
+
+    /**
+     * Triggers a change in the tab group: Clearing the preview selection
+     */
+    public onTabChange(): void {
+        this.importer.clearPreview();
+    }
+
+    /**
+     * Initializes the table
+     */
+    public initTable(): void {
+        const entryObservable = this.importer.getNewEntries();
+        this.subscriptions.push(
+            entryObservable.pipe(distinctUntilChanged(), auditTime(100)).subscribe(newEntries => {
+                this.hasFile = newEntries.length > 0;
+            })
```

Comment:
`this.hasFile` way exclusively accessed with `this.importer`, this checking for `this.hasFile` seems neglectable (or should be inside an own function with `this.importer`).

Since audit time seems relevant somehow:
```ts
this.hasFile: Observable<boolean> = this.importer.getNewEntries().pipe(
            distinctUntilChanged(),
            auditTime(100),
            map(entries => entries.length > 0)
        );
```
the template can then have `hasFile | async` which might be sufficient. nGrid  is then probably not necessary anymore.